hardhat.config.ts

```diff
@@ -0,0 +1,142 @@
+import '@nomiclabs/hardhat-ethers'
+import '@nomiclabs/hardhat-solhint'
+import '@nomiclabs/hardhat-truffle5'
+import '@nomiclabs/hardhat-waffle'
+import dotenv from 'dotenv'
+import { existsSync } from 'fs'
+import fs from 'fs/promises'
+import 'hardhat-abi-exporter'
+import 'hardhat-deploy'
+import 'hardhat-gas-reporter'
+import { HardhatUserConfig, task } from 'hardhat/config'
+
+const archivedDeploymentPath = './deployments/archive'
+
+// Load environment variables from .env file. Suppress warnings using silent
+// if this file is missing. dotenv will never modify any environment variables
+// that have already been set.
+// https://github.com/motdotla/dotenv
+dotenv.config({ debug: false })
+
+task('accounts', 'Prints the list of accounts', async (_, hre) => {
+  const accounts = await hre.ethers.getSigners()
+
+  for (const account of accounts) {
+    console.log(account.address)
+  }
+})
+
+task('save', 'Saves a specified contract as a deployed contract')
```

Comment:
Wouldn't it make more sense to make this command save *all* the contracts from the current deployment files, if they've been modified since the previous snapshot? Then you don't need to supply any arguments and are less likely to miss something.

Reply:
Since it's mostly meant to be run alongside a deployment, saving all the contracts probably isn't necessary since the archive is mostly used for the bytecode and ABI of the given deployed contract. If you were to save all the contracts, the snapshot might not be accurate with what is actually deployed on the network.

Reply:
Right, but the goal is to have an accurate snapshot of past versions of contracts, correct? Rather than asking people to remember to save individual contracts as they're changed, we should probably have a function that just makes sure there's a snapshot of the current version of each contract stored, and that can be run idempotently.

Reply:
The goal is to have an accurate snapshot of previous deployed contracts, if a contract isn't actually deployed from the snapshot I don't think it's particularly useful.